smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/waiters/KotlinJmespathExpressionVisitor.kt

```diff
@@ -71,11 +71,20 @@ class KotlinJmespathExpressionVisitor(val writer: KotlinWriter) : ExpressionVisi
     }
 
     override fun visitAnd(expression: AndExpression): String {
-        throw CodegenException("AndExpression is unsupported")
+        writer.addImport(RuntimeTypes.Utils.truthiness)
```

Comment:
No `#T` formatter in the writer?

Reply:
No, the existing `addTempVar` method accepts a literal code block and doesn't accept/pass placeholder args down to the writer. I could expand it but we've already pushed the scope of this PR far more than I'd like. I'd prefer to keep it simple for now.